audb/core/publish.py

```diff
@@ -127,7 +169,29 @@ def job(file):
     if add_media:
         deps._add_media(add_media)
```

Comment:
The calls to `deps._add_media()` and `deps._update_media()` will this affect `deps` only inside this function, or will it affect the `deps` object used in the `publish()` function as well?

Reply:
The latter. We basically update the old dependencies to the new version, which we later upload.

Reply:
Could you then update the docstring or comment before this section that it also updates the `deps` object outside of this function.

Reply:
Really? I mean, the function does not have a return value so what else should it do? I mean, we do it all the time in other places, too. E.g. `deps._update_media_version()`, `deps._drop()`, etc.

Reply:
Because with very few exceptions like `int` or `float', values in Python are always passed by reference and not as copy, e.g.:

```python
d = {}

def func(d):
    d['foo'] = 'bar'

func(d)
print(d)
```
```
{'foo': 'bar'}
```

If we would do:

```
deps = deps._add_media(...)
```

then I would expect that we receive a copy of `deps`, i.e. the original object passed to the function remains unchanged. But if for some reason it would not, then I would add a comment.

Reply:
Ok, that makes sense. It's now mentioned in the docstring of `_find_tables` and `_find_media` that ``deps`` is changed.

Reply:
Cool, thanks.